Homebrew install fails during make

**Describe the bug**
Installing the repository via `brew` fails at `make`. This issue appears to [have been resolved](https://github.com/stefanhaustein/TerminalImageViewer/issues/76), but now it seems to be back.

```zsh
==> Installing tiv from stefanhaustein/terminalimageviewer
==> Downloading https://github.com/stefanhaustein/TerminalImageViewer/archive/v1.1.0.tar.gz
Already downloaded: ~/Library/Caches/Homebrew/downloads/ac285ddaa9d98e1d87ce2e1cda1004d1ee572aceff2f1289dd1f54549858153c--TerminalImageViewer-1.1.0.tar.gz
==> make
Last 15 lines from ~/Library/Logs/Homebrew/tiv/01.make:
2021-07-14 23:02:11 -0400

make

clang++ -O2 -fpermissive -std=c++17 -Wall -fexceptions   -c -o tiv.o tiv.cpp
clang++ -pthread tiv.o -o tiv  -lstdc++fs
ld: library not found for -lstdc++fs
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [tiv] Error 1

If reporting this issue please do so to (not Homebrew/brew or Homebrew/core):
  stefanhaustein/terminalimageviewer

Traceback (most recent call last):
	29: from /usr/local/Homebrew/Library/Homebrew/build.rb:229:in '<main>'
	28: from /usr/local/Homebrew/Library/Homebrew/build.rb:133:in 'install'
	27: from /usr/local/Homebrew/Library/Homebrew/utils.rb:550:in 'with_env'
	26: from /usr/local/Homebrew/Library/Homebrew/build.rb:138:in 'block in install'
	25: from /usr/local/Homebrew/Library/Homebrew/formula.rb:1260:in 'brew'
	24: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2375:in 'stage'
	23: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/forwardable.rb:230:in 'stage'
	22: from /usr/local/Homebrew/Library/Homebrew/resource.rb:91:in 'stage'
	21: from /usr/local/Homebrew/Library/Homebrew/resource.rb:116:in 'unpack'
	20: from /usr/local/Homebrew/Library/Homebrew/resource.rb:199:in 'mktemp'
	19: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in 'run'
	18: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in 'chdir'
	17: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in 'block in run'
	16: from /usr/local/Homebrew/Library/Homebrew/resource.rb:117:in 'block in unpack'
	15: from /usr/local/Homebrew/Library/Homebrew/download_strategy.rb:102:in 'stage'
	14: from /usr/local/Homebrew/Library/Homebrew/download_strategy.rb:115:in 'chdir'
	13: from /usr/local/Homebrew/Library/Homebrew/download_strategy.rb:115:in 'chdir'
	12: from /usr/local/Homebrew/Library/Homebrew/resource.rb:121:in 'block (2 levels) in unpack'
	11: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2395:in 'block in stage'
	10: from /usr/local/Homebrew/Library/Homebrew/utils.rb:550:in 'with_env'
	 9: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2396:in 'block (2 levels) in stage'
	 8: from /usr/local/Homebrew/Library/Homebrew/formula.rb:1267:in 'block in brew'
	 7: from /usr/local/Homebrew/Library/Homebrew/build.rb:178:in 'block (2 levels) in install'
	 6: from /usr/local/Homebrew/Library/Taps/stefanhaustein/homebrew-terminalimageviewer/tiv.rb:11:in 'install'
	 5: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:128:in 'cd'
	 4: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:128:in 'chdir'
	 3: from /usr/local/Homebrew/Library/Taps/stefanhaustein/homebrew-terminalimageviewer/tiv.rb:12:in 'block in install'
	 2: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2166:in 'system'
	 1: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2166:in 'open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2230:in 'block in system': Failed executing: make (BuildError)
	9: from /usr/local/Homebrew/Library/Homebrew/brew.rb:155:in '<main>'
	8: from /usr/local/Homebrew/Library/Homebrew/brew.rb:167:in 'rescue in <main>'
	7: from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:509:in 'dump'
	6: from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:455:in 'issues'
	5: from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:459:in 'fetch_issues'
	4: from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:61:in 'issues_for_formula'
	3: from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:30:in 'search_issues'
	2: from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:160:in 'search'
	1: from /usr/local/Homebrew/Library/Homebrew/utils/github/api.rb:234:in 'open_rest'
/usr/local/Homebrew/Library/Homebrew/utils/github/api.rb:304:in 'raise_error': Validation Failed: [{"message"=>"The listed users and repositories cannot be searched either because the resources do not exist or you do not have permission to view them.", "resource"=>"Search", "field"=>"q", "code"=>"invalid"}] (GitHub::API::ValidationFailedError)

```


**To Reproduce**

Install `tiv` on `brew`, after tapping `brew tap stefanhaustein/TerminalImageViewer https://github.com/stefanhaustein/TerminalImageViewer`

Error occurs during `make`.


**OS and Dependency Information**
- MacOS `11.4` (Big Sur)
- 64 bit Intel Macbook Pro `x86_64-apple-darwin20.0` (non-Apple Silicon)
- Homebrew `3.2.2`
- zsh `5.8`
- Apple clang version 12.0.5 `clang-1205.0.22.9`

--- a/terminalimageviewer.rb
+++ b/terminalimageviewer.rb
@@ -4,6 +4,7 @@ class Terminalimageviewer < Formula
   url "https://github.com/stefanhaustein/TerminalImageViewer/archive/v1.1.0.tar.gz"
   sha256 "727ddaa048643cb0a13e23bc308415060d99d8b1853883ad339ad7b952050d61"
   head "https://github.com/stefanhaustein/TerminalImageViewer.git"
+  uses_from_macos "curl" => :test
   depends_on "gcc" => :build unless OS.linux?
   depends_on "imagemagick"
 
@@ -30,8 +31,10 @@ class Terminalimageviewer < Formula
 
   test do
     # Downloads a public domain test file from wikimedia commons and displays it.
-    # For some reason, when you redirect the output it is blank.
-    assert_equal "",
-    shell_output("#{bin}/tiv -0 https://upload.wikimedia.org/wikipedia/commons/2/24/Cornell_box.png").strip
+    # NOTE: Test will fail. Will fix when I get to my mac.
+    system "curl -o \"test.png\" https://upload.wikimedia.org/wikipedia/commons/2/24/Cornell_box.png"
+    assert_equal "[48;2;0;0;0m[38;2;0;0;0m▄[38;2;26;15;8m▄[38;2;39;28;16m▄[38;2;42;29;17m▄[38;2;41;29;17m▄[38;2;43;30;17m▄[38;2;44;30;17m▄[38;2;43;30;17m▄[38;2;44;31;18m▄[38;2;45;31;18m▄[38;2;46;32;18m▄▄[38;2;46;33;19m▄▄▄[38;2;47;34;19m▄▄[38;2;49;35;20m▄[38;2;49;36;20m▄[38;2;47;35;20m▄[38;2;48;35;20m▄[38;2;49;36;21m▄[38;2;49;37;21m▄▄[38;2;50;38;22m▄[38;2;51;39;22m▄[38;2;50;38;22m▄[38;2;49;37;21m▄[38;2;49;38;22m▄[38;2;50;39;22m▄[38;2;49;38;22m▄▄[38;2;50;39;22m▄[38;2;49;39;22m▄[38;2;50;39;22m▄[38;2;51;40;23m▄[38;2;49;39;22m▄[38;2;50;40;23m▄[38;2;49;39;22m▄[38;2;50;40;23m▄▄▄[38;2;49;40;23m▄[38;2;51;41;23m▄[38;2;49;40;23m▄▄[38;2;49;40;22m▄[38;2;48;40;22m▄[38;2;49;40;23m▄[38;2;49;40;22m▄[38;2;49;40;23m▄[38;2;47;39;22m▄▄▄▄[38;2;46;39;21m▄[38;2;45;38;21m▄[38;2;46;38;21m▄[38;2;45;37;21m▄[38;2;43;36;20m▄[38;2;44;38;20m▄[38;2;42;36;20m▄[38;2;42;35;19m▄▄[38;2;41;35;19m▄▄[38;2;40;34;18m▄[38;2;39;33;18m▄[38;2;40;34;18m▄▄[38;2;38;32;18m▄[38;2;38;32;17m▄[38;2;37;31;17m▄[38;2;22;21;10m▄[38;2;0;0;0m▄[0m
+",
+    shell_output("#{bin}/tiv -0 -w 400 -h 400 test.png").strip
   end
 end

--- a/terminalimageviewer.rb
+++ b/terminalimageviewer.rb
@@ -11,7 +11,7 @@ class Terminalimageviewer < Formula
     cd "src/main/cpp" do
       system "make"
       if OS.mac?
-        # No expermimental/filesystem.h on mac. 
+        # No expermimental/filesystem.h on mac.
         system "/usr/local/bin/g++-#{Formula["gcc"].version_suffix}", "-std=c++17",
                                                                                   "-Wall",
                                                                                   "-fpermissive",
@@ -19,9 +19,8 @@ class Terminalimageviewer < Formula
                                                                                   "-O2", "-c",
                                                                                   "-L/usr/local/opt/gcc/lib/gcc/11/",
                                                                                   "tiv.cpp", "-o", "tiv.o"
-        system "/usr/local/bin/g++-#{Formula["gcc"].version_suffix}", "tiv.o", "-o", 
-                                                                                 "tiv", "-L/usr/local/opt/gcc/lib/gcc/11/",
-                                                                                 "-pthread", "-s"
+        system "/usr/local/bin/g++-#{Formula["gcc"].version_suffix}", "tiv.o", "-o", "tiv",
+"-L/usr/local/opt/gcc/lib/gcc/11/", "-pthread", "-s"
       else
         system "make"
       end

--- a/tiv.rb
+++ b/tiv.rb
@@ -16,6 +16,7 @@ class Tiv < Formula
                                                                                  "-s"
       else
         system "make"
+      end
       bin.install "tiv"
     end
   end

--- a/tiv.rb
+++ b/tiv.rb
@@ -9,7 +9,13 @@ class Tiv < Formula
 
   def install
     cd "src/main/cpp" do
-      system "make"
+      if OS.mac?
+        system "c++", "-std=c++17", "-Wall", "-fpermissive", "-fexceptions", "-O2",
+                                                                                  "-c", "tiv.cpp", "-o", "tiv.o"
+        system "c++", "tiv.o", "-o", "tiv", "-L/usr/local/opt/gcc/lib/gcc/11/", "-pthread",
+                                                                                 "-s"
+      else
+        system "make"
       bin.install "tiv"
     end
   end

--- a/tiv.rb
+++ b/tiv.rb
@@ -4,7 +4,7 @@ class Tiv < Formula
   url "https://github.com/stefanhaustein/TerminalImageViewer/archive/v1.1.0.tar.gz"
   sha256 "727ddaa048643cb0a13e23bc308415060d99d8b1853883ad339ad7b952050d61"
   head "https://github.com/stefanhaustein/TerminalImageViewer.git"
-  depends_on "gcc" => :build unless OS.linux?
+  uses_from_macos "curl" => :test
   depends_on "imagemagick"
 
   def install
@@ -24,7 +24,9 @@ class Tiv < Formula
   test do
     # Downloads a public domain test file from wikimedia commons and displays it.
     # NOTE: Test will fail. Will fix when I get to my mac.
-    assert_equal "",
-    shell_output("#{bin}/tiv -0 -w 400 -h 400 https://upload.wikimedia.org/wikipedia/commons/2/24/Cornell_box.png").strip
+    system "curl -o \"test.png\" https://upload.wikimedia.org/wikipedia/commons/2/24/Cornell_box.png"
+    assert_equal "[48;2;0;0;0m[38;2;0;0;0m▄[38;2;26;15;8m▄[38;2;39;28;16m▄[38;2;42;29;17m▄[38;2;41;29;17m▄[38;2;43;30;17m▄[38;2;44;30;17m▄[38;2;43;30;17m▄[38;2;44;31;18m▄[38;2;45;31;18m▄[38;2;46;32;18m▄▄[38;2;46;33;19m▄▄▄[38;2;47;34;19m▄▄[38;2;49;35;20m▄[38;2;49;36;20m▄[38;2;47;35;20m▄[38;2;48;35;20m▄[38;2;49;36;21m▄[38;2;49;37;21m▄▄[38;2;50;38;22m▄[38;2;51;39;22m▄[38;2;50;38;22m▄[38;2;49;37;21m▄[38;2;49;38;22m▄[38;2;50;39;22m▄[38;2;49;38;22m▄▄[38;2;50;39;22m▄[38;2;49;39;22m▄[38;2;50;39;22m▄[38;2;51;40;23m▄[38;2;49;39;22m▄[38;2;50;40;23m▄[38;2;49;39;22m▄[38;2;50;40;23m▄▄▄[38;2;49;40;23m▄[38;2;51;41;23m▄[38;2;49;40;23m▄▄[38;2;49;40;22m▄[38;2;48;40;22m▄[38;2;49;40;23m▄[38;2;49;40;22m▄[38;2;49;40;23m▄[38;2;47;39;22m▄▄▄▄[38;2;46;39;21m▄[38;2;45;38;21m▄[38;2;46;38;21m▄[38;2;45;37;21m▄[38;2;43;36;20m▄[38;2;44;38;20m▄[38;2;42;36;20m▄[38;2;42;35;19m▄▄[38;2;41;35;19m▄▄[38;2;40;34;18m▄[38;2;39;33;18m▄[38;2;40;34;18m▄▄[38;2;38;32;18m▄[38;2;38;32;17m▄[38;2;37;31;17m▄[38;2;22;21;10m▄[38;2;0;0;0m▄[0m
+",
+    shell_output("#{bin}/tiv -0 -w 400 -h 400 test.png").strip
   end
 end

